lib/aff4_objects/security.py

```diff
@@ -151,10 +151,11 @@ class ApprovalWithApproversAndReason(Approval):
 
   class SchemaCls(Approval.SchemaCls):
     """The Schema for the ClientAccessApproval class."""
+
     LIFETIME = aff4.Attribute(
         "aff4:approval/lifetime", rdfvalue.RDFInteger,
-        "The number of microseconds an approval is valid for.",
-        default=4 * 7 * 24 * 60 * 60 * 1000000)  # 4 weeks
+        "The number of seconds an approval is valid for.",
```

Comment:
We do use this lifetime (at least we try, I'm not sure it actually works) in CreateApprovalRaw, it would make sense to keep this functionality. Why don't we set the default here to 0 and below we do

lifetime = self.Get(self.Schema.LIFETIME) or config_lib.CONFIG["ACL.token_expiry"]

That way we can have the best of both worlds.

Also, I agree that this should be kept in microseconds everywhere, adding conversions somewhere in the middle of the code just before we use the value seems a bit error prone.